src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -333,6 +333,45 @@ public override async Task WaitForSidecarAsync(CancellationToken cancellationTok
             }
         }
 
+        public override async Task<DaprMetadata> GetMetadataAsync(CancellationToken cancellationToken = default)
+        {
+            var path = "/v1.0/metadata";
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri(this.httpEndpoint, path));
+
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
+                response.EnsureSuccessStatusCode();
+
+                return await response.Content.ReadFromJsonAsync<DaprMetadata>(this.jsonSerializerOptions, cancellationToken);
+            }
+            catch (HttpRequestException ex)
+            {
+                throw new DaprException("Get metadata operation failed: the Dapr endpoint indicated a failure. See InnerException for details.", ex);
+            }
+        }
+
+        public override async Task SetMetadataAsync(string attributeName, string attributeValue, CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(attributeName, nameof(attributeName));
+
+            var path = $"/v1.0/metadata/{attributeName}";
+            var request = new HttpRequestMessage(HttpMethod.Put, new Uri(this.httpEndpoint, path))
+            {
+                Content = new StringContent(attributeValue)
+            };
+
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
+                response.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException ex)
+            {
+                throw new DaprException("Set metadata operation failed: the Dapr endpoint indicated a failure. See InnerException for details.", ex);
+            }
+        }
```

Comment:
Both of these should use the internal client. The metadata API is specified in the proto, so we should use it's generated client.

Reply:
The grpc API is currently unavailable If grpc API is used, it is not available in the lower version of daprd, see https://github.com/dapr/dapr/issues/4941

Reply:
Does it need to be modified here? If need to modify, I will revert to the first [commit](https://github.com/dapr/dotnet-sdk/pull/917/commits/48fb236c8649f02c1ae1c637d3f9aea4be45fd7a)